crates/rpc/rpc-eth-api/src/helpers/transaction.rs

```diff
@@ -643,3 +646,58 @@ pub trait LoadTransaction: SpawnBlocking {
         }
     }
 }
+
+/// Find a transaction by sender and nonce
+pub async fn get_transaction_by_sender_and_nonce<Eth>(
+    eth: &Eth,
+    sender: Address,
+    nonce: u64,
+    include_pending: bool,
+) -> Result<Option<RpcTransaction<Eth::NetworkTypes>>, Eth::Error>
+where
+    Eth: EthBlocks + EthState,
+{
+    // Check the pool first
+    if include_pending {
+        if let Some(tx) = LoadState::pool(eth).get_transaction_by_sender_and_nonce(sender, nonce) {
+            let transaction = tx.transaction.clone().into_consensus();
+            return Ok(Some(from_recovered(transaction)));
+        }
+    }
+
+    // Check if the sender is a contract
+    if EthState::get_code(eth, sender, None).await?.len() > 0 {
```

Comment:
```suggestion
    if LoadState::get_code(eth, sender, None).await?.len() > 0 {
```
add `get_code` to `LoadState`.

the thought behind the `Load` and `Eth` prefixed traits is: if more than one `Eth` trait needs to call a method, it goes in the `Load` trait. No `Eth` trait method should require another `Eth` trait.